From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Tue, 25 Dec 2018 13:10:18 +0900
Subject: switch-root: fix error message

Fixes #11261.

(cherry picked from commit a5c67ccc575e6ebf12710cb7df84f65a51c5dc58)
(cherry picked from commit ebcd154e1df434865d2752efdccbc7737bb28029)
---
 src/shared/switch-root.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/switch-root.c b/src/shared/switch-root.c
index ee31c44..dbb4622 100644
--- a/src/shared/switch-root.c
+++ b/src/shared/switch-root.c
@@ -83,7 +83,7 @@ int switch_root(const char *new_root,
                         (void) mkdir_p_label(chased, 0755);
 
                 if (mount(i, chased, NULL, mount_flags, NULL) < 0)
-                        return log_error_errno(r, "Failed to mount %s to %s: %m", i, chased);
+                        return log_error_errno(errno, "Failed to mount %s to %s: %m", i, chased);
         }
 
         /* Do not fail if base_filesystem_create() fails. Not all switch roots are like base_filesystem_create() wants
